### ISSUE TITLE
Stepper content states not announcing to screen readers

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3529]
## Describe the bug

When a stepper is in an error or success state, the state is only represented with color/icon and not announced (such as "completed"). Also, there are cases where the panel cannot be closed (step 1 which is not completed), and should be marked as such.

## How to reproduce

https://clarity.design/documentation/stepper

- Using as screen reader, go to the stepper example
- Trigger an error message by using the Tab key to focus the 'First Name' field and then press the Tab key again to move focus to the next control in the Tab order.
- Observe that an error icon has appeared within the "clr-accordion-header-button" button 
- Tab to the "clr-accordion-header-button" button and observe that you do not hear any information about the application being in an error state when the read this button
- Same for success

- Navigate to a stepper with a screen reader and try to close a step that is not closable, get confusing messaging about expandable state

## Expected behavior

- Create a textual error/success message to be used with aria-describedby to programmatically associate the error/success message with the "clr-accordion-header-button" button.
- Set `disabled` state on clr-accordion-header-button when state cannot be toggled (even if open).

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

_Add any other notes about the problem here._
